applications/KWS_Phoneme/README.md

```diff
@@ -0,0 +1,51 @@
+
+In the second part, we use the predicted phoneme outputs from the phoneme classifier for predicting the input keyword. We train a 1 layer FastGRNN classifier to predict the keyword based on the phoneme transcription as input. Since the phoneme classifier training has been done to account for diverse accent, background noise and environments, the keyword classifier can be trained using a small number of Text-To-Speech(TTS) samples generated using any standard TTS api from cloud services like Azure, Google Cloud or AWS.
+
+This gives two advantages: (a) The phoneme model is trained to account for diverse accents and background noise settings, thus the flexible keyword classifier training requires only a small number of keyword samples, and (b) Empirically this method was able to detect keywords from as far as 9ft of distance. Further, the phoneme model has a small size of around 250k parameters and can fit on a Cortex M7 micro-controller.
+
+# Training the Phoneme Classifier
+1) Train a phoneme classification model on some public speech dataset like LibriSpeech
+2) Training speech dataset can be labelled using Montreal Force Aligner
+3) Speech snippets are convolved with reverberation files, and additive noises from YouTube or other open source are added
+4) We also add white gaussian noise of various SNRs
+
+# Training the KWS Model
+1) Our method takes as input the speech snippet and passes it through the phoneme classifier
+2) Keywords are detected by training a keyword classifier over the detected phonemes
+3) For training the keyword classifier, we use Azure and Google Text-To-Speech API to get the training data (keyword snippets)
+4) For example, if you want to train a Keyword classifier for the keywords in the Google30 dataset, generate TTS samples from the Azure/Google-Cloud/AWS API for each of the 30 keywords. The TTS samples for each keyword must be stored in a separate folder named according to the keyword. More details about how the generated TTS data should be stored are mentioned below in sample use case for classifier model training.
+
+# Sample Use Cases
+
+## Phoneme Model Training
+The following command can be used to instantiate and train the phoneme model.
+```
+python train_phoneme.py --base_path=/path/to/librispeech_data/ --rir_base_path=/path/to/reverb_files/ --additive_base_path=/path/to/additive_noises/ --snr_samples="0,5,10,25,100,100" --rir_chance=0.5 
+```
+Some important command line arguments:
+1) base_path : Path of the speech data folder. The data in this folder should be in accordance to the dataloader code written here. 
+2) rir_base_path, additive_base_path : Path to the reverb and additive noise files
+3) snr_samples : List of various SNRs at which the additive noise is to be added.
+4) rir_chance : Probability at which reverberation has to be done for each speech sample
```

Comment:
What is the meaning of `Probability at which reverberation has to be done`? Also, please make use of full stops everywhere.